app/src/main/java/at/roteskreuz/stopcorona/model/repositories/InfectionMessengerRepository.kt

```diff
@@ -203,10 +201,23 @@ class InfectionMessengerRepositoryImpl(
         downloadMessagesStateObserver.loading()
         withContext(coroutineContext) {
             try {
-                val archives: List<File> = fetchBatchDiagnosisKeysBasedOnInfectionLevel(warningType)
-                val contextToken = BATCH + "," + warningType + "," + UUID.randomUUID().toString()
+                val warningType = quarantineRepository.getCurrentWarningType()
+                val contextToken = UUID.randomUUID().toString()
+                val index = apiInteractor.getIndexOfDiagnosisKeysArchives()
+                val fullBatchParts = fetchFullBatchDiagnosisKeys(index.fullBatchForWarningType(warningType))
+                val dailyBatchesParts = fetchDailyBatchesDiagnosisKeys(index.dailyBatches)
```

Comment:
check if needed when green @falkorichter

Reply:
I will continue here, but this is more relevant regarding the other decision tree branches. It´s not critical to optimize this so early.

Reply:
@falkorichter please take note